### PR TITLE
Remove the instantiate button for SSP

### LINKED ISSUE #6258: Placeholder

### PR DESCRIPTION
**Pull request by adeas31**
Fixes #6258

### CODE REVIEW

**Comment by adeas31** [comment]
@lochel and @AnHeuermann the example provided in the ticket works fine with these changes. Can you guys test some more before we push this in?

I removed the instantiate button as is not needed separately. When the user simulates the model then `instantiate`, `initialize`, `simulate` and `terminate` is called.

@lochel I also removed the code that checks the model state. Since `terminate` is called both in success and failure cases and I trust that it clears/sets the model state properly.

**Review by lochel — APPROVED** [review]
I tested it with some other models and it works fine for me. I can simulate, resimulate and change parameters/start-values. 👍

**Review by AnHeuermann — APPROVED** [review]
Great. Thanks!
Now I can simulate, change values, re-simulate.
My example I want to present in MODPROD is now working as well.

File: OMEdit/OMEditLIB/MainWindow.cpp [hunk 1]
```diff
@@ -2340,8 +2286,9 @@ void MainWindow::simulateModelWithTransformationalDebugger()
 }
 
 /*!
-  Simualtes the model with algorithmic debugger
-  */
+ * \brief MainWindow::simulateModelWithAlgorithmicDebugger
+ * Simualtes the model with algorithmic debugger
```
Comment (adrpo) [inline OMEdit/OMEditLIB/MainWindow.cpp:2290]: Typo: Simualtes

File: OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp [hunk 2]
```diff
@@ -91,28 +91,33 @@ OMSSimulationOutputWidget::OMSSimulationOutputWidget(const QString &cref, QWidge
...
+        mpCancelSimulationButton->setEnabled(true);
+      } else {
+        mpProgressLabel->setText(tr("Simulation using the <b>%1</b> model is failed. %2").arg(mCref).arg(GUIMessages::getMessage(GUIMessages::CHECK_MESSAGES_BROWSER)));
+        mpProgressBar->setValue(mpProgressBar->maximum());
+        mpArchivedOMSSimulationItem->setStatus(tr("Simulation failed!"));
+        OMSProxy::instance()->terminate(mCref);
+      }
     } else {
-      mpProgressLabel->setText(tr("Simulation using the <b>%1</b> model is failed. %2").arg(mCref).arg(GUIMessages::getMessage(GUIMessages::CHECK_MESSAGES_BROWSER)));
+      mpProgressLabel->setText(tr("Initialization using the <b>%1</b> model is failed. %2").arg(mCref).arg(GUIMessages::getMessage(GUIMessages::CHECK_MESSAGES_BROWSER)));
```
Comment (adrpo) [inline OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp:111]: Sounds weird, maybe: "Initialization of the model <b>%1</b> has failed: %2"

File: OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp [hunk 3]
```diff
@@ -91,28 +91,33 @@ OMSSimulationOutputWidget::OMSSimulationOutputWidget(const QString &cref, QWidge
...
+      OMSProxy::instance()->terminate(mCref);
     }
   } else {
-    LibraryTreeItem *pLibraryTreeItem;
-    pLibraryTreeItem = MainWindow::instance()->getLibraryWidget()->getLibraryTreeModel()->findLibraryTreeItemOneLevel(mCref);
-    if (pLibraryTreeItem) {
-      MainWindow::instance()->instantiateOMSModel(pLibraryTreeItem, false);
-    }
-    mpProgressLabel->setText(tr("Initialization using the <b>%1</b> model is failed. %2").arg(mCref).arg(GUIMessages::getMessage(GUIMessages::CHECK_MESSAGES_BROWSER)));
+    mpProgressLabel->setText(tr("Instantiation using the <b>%1</b> model is failed. %2").arg(mCref).arg(GUIMessages::getMessage(GUIMessages::CHECK_MESSAGES_BROWSER)));
```
Comment (adrpo) [inline OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp:117]: Same as above: "Instantiation of the model %1 has failed: %2"

**Comment by adrpo** [comment]
Also, one needs to change the documentation to remove the initialization step I guess.

File: OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp (hunk 2 above)
Comment (adeas31) [reply OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp:111]: Isn't `Initialization of the model %1 is failed: %2` is more appropriate? Then we have sentences like `Simulation using the <b>%1</b> model is cancelled.` and `Simulation using the <b>%1</b> model is finished.` which should also be updated like `Simulation of the model <b>%1</b> is cancelled.` and `Simulation of the model <b>%1</b> is finished.`

Comment (adrpo) [reply OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp]: Now is worse :)
These are actions performed on the model so not "using" and not "is", the action has been done it has failed or worked so we need to use "has".

I would go for "Initialization/Simulation of model X has failed. msg"

Comment (adrpo) [reply OMEdit/OMEditLIB/OMS/OMSSimulationOutputWidget.cpp]: Ah, i now saw you changed using to of the model, so is better. I guess I should stop looking at PRs Friday night :)
Also "Simulation using the <b>%1</b> model is cancelled." is not ok, I would go for "Simulation of model <b>%1</b> has been cancelled."

But I'm no expert in English so maybe we should ask somebody else, @sjoelund any comments?

**Pull request merged**